deploy/osd-25526-oao-olm-dance/01-cronjob.yaml

```diff
@@ -0,0 +1,64 @@
+---
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: sre-operator-reinstall
+  namespace: openshift-ocm-agent-operator
+spec:
+  ttlSecondsAfterFinished: 100
+  failedJobsHistoryLimit: 1
+  successfulJobsHistoryLimit: 3
+  concurrencyPolicy: Replace
+  schedule: "* * * * *"
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          serviceAccountName: sre-operator-reinstall-sa
+          restartPolicy: Never
+          containers:
+          - name: operator-reinstaller
+            image: image-registry.openshift-image-registry.svc:5000/openshift/cli:latest
+            imagePullPolicy: Always
+            command:
+            - sh
+            - -c
+            - |
+              #!/bin/bash
+              set -euxo pipefail
+              NAMESPACE=openshift-ocm-agent-operator
+
+              # Check for the status OAO v0.1.289 version (previous version:
+              # ocm-agent-operator.v0.1.289-g742e13a). If we it's not succeeded,
+              # do OLM dance The resources will be re-created with the latest
+              # version
+              CSV_PHASE=$(oc -n openshift-ocm-agent-operator get csv ocm-agent-operator.v0.1.289-g742e13a -ojsonpath='{.status.phase}{"\n"}') || true
+              #CSV_STATUS=$(oc -n openshift-ocm-agent-operator get csv -l operators.coreos.com/ocm-agent-operator.openshift-ocm-agent-operator= -oname;echo $?)
+              # This does nothing if the expected CSV is not found.
+              if [[ -n ${CSV_PHASE} && ${CSV_PHASE} != "Succeeded" ]]; then
+                # Take backup of the files to be recreated (subscription, catalogsource and operatorgroup)
+                oc -n "$NAMESPACE" get subs ocm-agent-operator -oyaml > /tmp/01-ocm-agent-operator-subs.yaml
+                oc -n "$NAMESPACE" get catsrc ocm-agent-operator-registry -oyaml > /tmp/02-ocm-agent-operator-registry.yaml
+                oc -n "$NAMESPACE" get og ocm-agent-operator-og -oyaml > /tmp/03-ocm-agent-operator-og.yaml
+
+                oc -n "$NAMESPACE" delete managednotification --all
+                oc -n "$NAMESPACE" delete managedfleetnotifications --all
```

Comment:
I am not sure why we need to delete the CRs, but do we need to delete the CRD as well?

Reply:
> I am not sure why we need to delete the CRs, but do we need to delete the CRD as well?

Yeah, if the OAO can be reinstalled without deleting the CRs, I think it's safer to avoid deleting them